server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -60,7 +61,8 @@ public CoordinatorDynamicConfig(
       // coordinator console can not send array of strings in the update request.
       // See https://github.com/druid-io/druid/issues/3055
       @JsonProperty("killDataSourceWhitelist") Object killDataSourceWhitelist,
-      @JsonProperty("killAllDataSources") Boolean killAllDataSources
+      @JsonProperty("killAllDataSources") Boolean killAllDataSources,
+      @JsonProperty("maxSegmentsInQueue") Integer maxSegmentsInQueue
```

Comment:
Could you please add deserialization test, showing that deserialization of an old config creates an object with unbounded queue?